google_takeout_parser/time_utils.py

```diff
@@ -10,11 +11,17 @@ def parse_datetime_millis(d: Union[str, float, int]) -> datetime:
     return parse_datetime_sec(int(d) / 1000)
 
 
-def parse_json_utc_date(ds: str) -> datetime:
-    utc_naive = datetime.fromisoformat(ds.rstrip("Z"))
-    return utc_naive.replace(tzinfo=timezone.utc)
+if sys.version_info[:2] >= (3, 11):
+    # form 3.11, it supports parsing strings ending with Z
+    parse_json_utc_date = datetime.fromisoformat
```

Comment:
I grepped all takeout data I had for isoformat-ish timestamps, they all end up with Z, so that should work!